### DISCUSSION TITLE
一个关于Blog文章自动生成目录的问题

### DISCUSSION THREAD

**Discussion by TonyWara** [rbind/yihui#1481] [Q&A] [answered]
yihui学长你好👋，我是长涛

我想提问一个关于网站技术的问题。在使用网站主题yihui/hugo-ivy的情况下，要如何实现阅读.md格式的单篇文章时，网站可以在文章侧边自动生成章节目录，并且能够使用鼠标点击完成跳转的功能。

我目前有搜索到一些相关的教程，了解到目录获取以及目录生成的原理以及相关代码，但由于我没有编程基础，不知道需要在原始网站的什么文件内去进行编辑和修改。又想到也许不同的网站主题也会影响制作类似效果的方案，于是只好在这里向学长提问，希望能得到一些思路以及学习方向

再次感谢🙏🙏

**Comment by yihui** [comment, marked as answer]
目录有多种可能的实现办法，我给你发送的 PR 是一种可能：https://github.com/TonyWara/Fractal-Reading-Party-BLOG/pull/1 它会对任意文章自动生成目录，效果如：https://deploy-preview-1--fractalreadingparty.netlify.app/post/2023/04/30/ai-02/

但是有个问题：这个 hugo-ivy 主题的设计是双栏布局，而你想要的似乎是浮动侧边目录，而不是放在文章开头的固定目录。如果要浮动的话，还需要一些 CSS 改动，把页面布局改成三栏，目录放在右栏。对这个特定的主题来说，这个 CSS 改动并非两三分钟的工作，所以我就不想折腾了。

若浮动目录对你来说很重要，不妨考虑使用我的另一个 Hugo 主题：https://prose.yihui.org/about/

**Reply by TonyWara** [reply]
收到！真的非常感谢🙏！我觉得目前这个目录就足够了，我来尝试使用一下这个脚本。如果之后有时间，我还蛮想进阶学习一些关于CSS的相关概念的哈哈哈哈
